lib/widgets/layout/scrollable_time_table.dart

```diff
@@ -0,0 +1,140 @@
+import 'package:flutter/material.dart';
+import 'package:flutter_gen/gen_l10n/app_localizations.dart';
+import 'package:horizontal_data_table/horizontal_data_table.dart';
+import 'package:provider/provider.dart';
+import 'package:cabin_booking/constants.dart';
+import 'package:cabin_booking/model/cabin.dart';
+import 'package:cabin_booking/model/cabin_manager.dart';
+import 'package:cabin_booking/model/day_handler.dart';
+import 'package:cabin_booking/widgets/layout/time_column.dart';
+import 'package:cabin_booking/widgets/booking/bookings_table.dart';
+import 'package:cabin_booking/widgets/cabin/cabin_icon.dart';
+
+class ScrollableTimeTable extends StatelessWidget {
+  const ScrollableTimeTable();
+  static const int sortName = 0;
+  static const int sortStatus = 1;
+  final bool isAscending = true;
+  final int sortType = sortName;
+
+  @override
+  Widget build(BuildContext context) {
+    return Expanded(
+      child: Container(
+        width: MediaQuery.of(context).size.width,
+        child: _getBodyWidget(context),
+      ),
+    );
+  }
+
+  Widget _getBodyWidget(context) {
+    final theme = Theme.of(context);
+
+    return LayoutBuilder(builder: (context, constraints) {
+      return Container(
+        color: theme.dialogBackgroundColor,
+        child: Consumer2<DayHandler, CabinManager>(
+          builder: (context, dayHandler, cabinManager, child) {
+            final maxParentWidth = constraints.maxWidth;
+            final calculatedBookingStackWidth =
+                (maxParentWidth - kTimeColumnWidth) /
+                    cabinManager.cabins.length;
+            final bookingStackWidth =
+                (calculatedBookingStackWidth < kBookingColumnMinWidth)
+                    ? kBookingColumnMinWidth
+                    : calculatedBookingStackWidth;
+
+            if (cabinManager.cabins.isEmpty) {
+              return Center(
+                child: Text(
+                  AppLocalizations.of(context).noCabins,
+                  textAlign: TextAlign.center,
+                  style: theme.textTheme.headline5
+                      .copyWith(color: Colors.grey[600]),
+                ),
+              );
+            }
+
+            return Container(
+              child: HorizontalDataTable(
+                leftHandSideColumnWidth: kTimeColumnWidth,
+                rightHandSideColumnWidth:
+                    bookingStackWidth * cabinManager.cabins.length,
+                isFixedHeader: true,
+                headerWidgets: _getTitleWidget(
+                    context, cabinManager.cabins, bookingStackWidth),
+                leftSideItemBuilder: _generateFirstColumnRow,
+                rightSideItemBuilder: _generateRightHandSideColumnRow,
+                itemCount: 1, // Hacking the widget a little bit
```

Comment:
Since widget make us split the table into separated cells, it limit us when we have multi-slot items.
Thus, we need to hack the widget a little bit and simulate we only have one table row that populates the 'fake' table rows (by using the same 'TimeColumn' and 'BookingsTable' widgets we already had) 😁